File: codewars/6/Sum of Digits - 541c8630095125aba6000c00/csharp/main.cs
public class Number
{
    public int DigitalRoot(long n)
    {
        if (n < 10) return (int) n;

        var sum = 0;
        while (n > 0) {
            sum += (int) (n % 10);
            n /= 10;
        }

        return DigitalRoot(sum);
    }
}

File: codewars/6/Sum of Digits - 541c8630095125aba6000c00/go/main.go
package kata

func DigitalRoot(n int) int {
	for ; n > 9; n = sum(n) {
	}

	return n
}

func sum(n int) (result int) {
	for ; n > 0; n /= 10 {
		result += n % 10
	}

	return result
}
